### PR TITLE
update traffic policy docs

### PR DESCRIPTION
**Pull request by Megalonia**

### CODE REVIEW

File: docs/http/traffic-policy/actions/redirect.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+| `redirect` |
+
+| Parameter     | &nbsp;                    | Description                                                                                                                                       |
+| ------------- | ------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `from`        | string                    | A regular expression string to match inside of the URL.                                                                                           |
+| `to`          | string                    | A regular expression string to replace the `from` match with. Also contains special [named variables](#named-variables) outside of regular regex. |
+| `status_code` | int                       | 3xx status code used for redirecting, 302 by default.                                                                                             |
+| `headers`     | Map&lt;string, string&gt; | Headers to be added to the request.                                                                                                               |
```
Comment (Megalonia) [inline docs/http/traffic-policy/actions/redirect.mdx:38]: should I add named variables?

Comment (rkolavo) [reply docs/http/traffic-policy/actions/redirect.mdx:38]: oooo yes I think so — it's also the same stuff reused right?

File: docs/http/traffic-policy/actions/add-headers.mdx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+## Example
+
+<ExampleHTTP />
+
+## Behavior
+
+When executed as an inbound policy, this action will add headers on an incoming http request before reaching the upstream server with
+the configured headers. When executed as an outbound policy, the configured headers are added to the response from the upstream server.
+
+<VariableInterpolation type="add-headers" />
```
Comment (Megalonia) [inline docs/http/traffic-policy/actions/add-headers.mdx:20]: should this be section after `### Configuration`?

Comment (Megalonia) [reply docs/http/traffic-policy/actions/add-headers.mdx]: going to go ahead and do this

File: docs/http/traffic-policy/actions/index.mdx [hunk 3]
```diff
@@ -9,11 +9,14 @@ pagination_label: Actions
...
+| [add-headrs](add-headers)          | Add headers from incoming requests or outgoing responses.                                | Inbound, Outbound |
+| [custom-response](custom-response) | Send back a pre-defined custom response to the client.                                   | Inbound, Outbound |
+| [deny](deny)                       | Reject incoming traffic to an endpoint.                                                  | Inbound           |
+| [jwt-validation](jwt-validation)   | Validate JSON Web Tokens (JWT).                                                          | Inbound           |
+| [log](log)                         | Emit an event with custom defined metadata.                                              | Inbound, Outbound |
+| [rate-limit](rate-limit)           | Rate limit traffic to your upstream servers.                                             | Inbound           |
+| [remove-headrs](remove-headers)    | Remove headers from incoming requests or outgoing responses.                             | Inbound, Outbound |
```
Comment (russorat) [inline docs/http/traffic-policy/actions/index.mdx:20]:
```suggestion
| [remove-headers](remove-headers)    | Remove headers from incoming requests or outgoing responses.                             | Inbound, Outbound |
```

File: docs/http/traffic-policy/actions/index.mdx [hunk 4]
```diff
@@ -9,11 +9,14 @@ pagination_label: Actions
...
-| [custom-response](custom-response) | Send back a pre-defined custom response to the client. | Inbound, Outbound |
-| [deny](deny)                       | Reject incoming traffic to an endpoint.                | Inbound           |
-| [jwt-validation](jwt-validation)   | Validate JSON Web Tokens (JWT).                        | Inbound           |
-| [log](log)                         | Emit an event with custom defined metadata.            | Inbound, Outbound |
-| [rate-limit](rate-limit)           | Rate limit traffic to your upstream servers.           | Inbound           |
-| [url-rewrite](url-rewrite)         | Transform request URLs using regex.                    | Inbound           |
+| Type                               | Description                                                                              | Supported On      |
+| ---------------------------------- | ---------------------------------------------------------------------------------------- | ----------------- |
+| [add-headrs](add-headers)          | Add headers from incoming requests or outgoing responses.                                | Inbound, Outbound |
```
Comment (russorat) [inline docs/http/traffic-policy/actions/index.mdx:14]:
```suggestion
| [add-headers](add-headers)          | Add headers from incoming requests or outgoing responses.                                | Inbound, Outbound |
```

File: docs/http/traffic-policy/actions/remove-headers.mdx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { ExampleHTTP } from "/examples/actions/remove-headers.mdx";
+
+# Remove Headers
+
+## Overview
+
+The Remove Headers policy action enables the removal of headers to both incoming requests and outbound responses.
```
Comment (russorat) [inline docs/http/traffic-policy/actions/remove-headers.mdx:7]: should we just say `inbound and outbound traffic`?

File: docs/http/traffic-policy/gallery.mdx [hunk 6]
```diff
@@ -72,3 +72,31 @@ This rule applies rate limiting of `30` requests per second to the endpoint `/ap
...
 />
+
+## User agent filtering
+
+Here we are serving custom content to Microsoft Edge users by checking whether the `User-Agent` header has case insensitive `(?i)` string `edg/` followed by some digits `\d`. You can test this by viewing this [regex101 link](https://regex101.com/r/3NPVub/1).
+
+Additionally, we have to escape the `\d` properly so it can be decoded from YAML / JSON. In YAML when your string is not wrapped in quotes you single-escape: `\\d` when it is wrapped in quotes you must double-escape: `\\\\d` for it to properly decode.
+
+<ConfigExample
+	config={{
```
Comment (russorat) [inline docs/http/traffic-policy/gallery.mdx:83]: should this be in the examples folder?

File: docs/http/traffic-policy/actions/remove-headers.mdx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+### Configuration
+
+| Type             |
+| ---------------- |
+| `remove-headers` |
+
+| Parameter | &nbsp;              | Description                                               |
+| --------- | ------------------- | --------------------------------------------------------- |
+| `headers` | Array&lt;string&gt; | Headers to be removed to be added to request or response. |
```
Comment (tammybailey) [inline docs/http/traffic-policy/actions/remove-headers.mdx:33]: I think the description text should be `Headers to be removed from the request or response.`

File: docs/http/traffic-policy/gallery.mdx [hunk 8]
```diff
@@ -72,3 +72,31 @@ This rule applies rate limiting of `30` requests per second to the endpoint `/ap
 		],
 	}}
 />
+
+## User agent filtering
+
+Here we are serving custom content to Microsoft Edge users by checking whether the `User-Agent` header has case insensitive `(?i)` string `edg/` followed by some digits `\d`. You can test this by viewing this [regex101 link](https://regex101.com/r/3NPVub/1).
+
+Additionally, we have to escape the `\d` properly so it can be decoded from YAML / JSON. In YAML when your string is not wrapped in quotes you single-escape: `\\d` when it is wrapped in quotes you must double-escape: `\\\\d` for it to properly decode.
```
Comment (tammybailey) [inline docs/http/traffic-policy/gallery.mdx:80]: Question: The first sentence says we need to escape the `\d` properly for YAML and JSON decoding. However, the second sentence starts with "In YAML". Is the text that follows true for YAML only or for JSON as well? If YAML only, we should also add information on what needs to be done for JSON.

Also, the second sentence should be two sentences, maybe something like ...
```
Strings that are not wrapped in quotes are single-escaped: `\\d`. Strings wrapped in quotes are double-escaped: `\\\\d`.
```

File: docs/http/traffic-policy/gallery.mdx [hunk 9]
```diff
@@ -72,3 +72,31 @@ This rule applies rate limiting of `30` requests per second to the endpoint `/ap
...
+
+<ConfigExample
+	config={{
+		inbound: [
+			{
+				expressions: [
+					"'User-Agent' in req.Headers",
+					"size(req.Headers['User-Agent'].filter(x, x.matches('(?i).*Edg/\\\\d+.*'))) > 0",
```
Comment (tammybailey) [inline docs/http/traffic-policy/gallery.mdx:88]: Verified this works as written in the UI

File: docs/http/traffic-policy/actions/index.mdx [hunk 10]
```diff
@@ -9,11 +9,14 @@ pagination_label: Actions
...
+| [custom-response](custom-response) | Send back a pre-defined custom response to the client.                                   | Inbound, Outbound |
+| [deny](deny)                       | Reject incoming traffic to an endpoint.                                                  | Inbound           |
+| [jwt-validation](jwt-validation)   | Validate JSON Web Tokens (JWT).                                                          | Inbound           |
+| [log](log)                         | Emit an event with custom defined metadata.                                              | Inbound, Outbound |
+| [rate-limit](rate-limit)           | Rate limit traffic to your upstream servers.                                             | Inbound           |
+| [remove-headrs](remove-headers)    | Remove headers from incoming requests or outgoing responses.                             | Inbound, Outbound |
+| [redirect](redirect)               | Transform your URL using Regular Expressions and redirect users through location header. | Inbound           |
```
Comment (tammybailey) [inline docs/http/traffic-policy/actions/index.mdx:21]: I don't think regular expressions should be capitalized here, although fine to be overruled.

File: docs/http/traffic-policy/gallery.mdx (hunk 8 above)
Comment (Megalonia) [reply docs/http/traffic-policy/gallery.mdx:80]: cc: @nijikokun

File: docs/http/traffic-policy/actions/index.mdx [hunk 11]
```diff
@@ -11,12 +11,12 @@ Traffic Policy actions enable you to modify the behavior of traffic flowing thro
...
 | [jwt-validation](jwt-validation)   | Validate JSON Web Tokens (JWT).                                                          | Inbound           |
 | [log](log)                         | Emit an event with custom defined metadata.                                              | Inbound, Outbound |
 | [rate-limit](rate-limit)           | Rate limit traffic to your upstream servers.                                             | Inbound           |
-| [remove-headers](remove-headers)    | Remove headers from incoming requests or outgoing responses.                             | Inbound, Outbound |
-| [redirect](redirect)               | Transform your URL using Regular Expressions and redirect users through location header. | Inbound           |
+| [remove-headers](remove-headers)   | Remove headers from incoming requests or outgoing responses.                             | Inbound, Outbound |
+| [redirect](redirect)               | Transform your URL using regular xxpressions and redirect users through location header. | Inbound           |
```
Comment (tammybailey) [inline docs/http/traffic-policy/actions/index.mdx:21]: Typo here: `xxpressions`

File: docs/http/traffic-policy/gallery.mdx (hunk 8 above)
Comment (nijikokun) [reply docs/http/traffic-policy/gallery.mdx:36]: JSON is always 4

**Review by nijikokun — APPROVED** [review]

**Pull request merged**